### ISSUE TITLE
ICE on linux with GCC 10.3

### ISSUE THREAD

**Issue by SylvainCorlay** [JuliaInterop/libcxxwrap-julia#105]
It seems that the release from a few days ago causes an internal compiler error with GCC 10.3 on linux-64.

cf https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=495914&view=logs&jobId=656edd35-690f-5c53-9ba3-09c10d0bea97

```
[ 41%] Building CXX object CMakeFiles/cxxwrap_julia_stl.dir/src/stl.cpp.o
during RTL pass: expand
In file included from /home/conda/feedstock_root/build_artifacts/libcxxwrap-julia_1650582518778/work/include/jlcxx/jlcxx.hpp:16,
                 from /home/conda/feedstock_root/build_artifacts/libcxxwrap-julia_1650582518778/work/src/stl.cpp:5:
/home/conda/feedstock_root/build_artifacts/libcxxwrap-julia_1650582518778/work/include/jlcxx/smart_pointers.hpp: In function 'int jlcxx::TypeWrapper<T>::apply_internal(FunctorT&&) [with AppliedT = std::unique_ptr<bool>; FunctorT = jlcxx::smartptr::WrapSmartPointer; T = jlcxx::Parametric<jlcxx::TypeVar<1> >]':
/home/conda/feedstock_root/build_artifacts/libcxxwrap-julia_1650582518778/work/include/jlcxx/smart_pointers.hpp:141:28: internal compiler error: Segmentation fault
  141 |     wrapped.module().method("__cxxwrap_smartptr_dereference", DereferenceSmartPointer<WrappedT>::apply);
      |     ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Please submit a full bug report,
```

**Comment by SylvainCorlay** [comment]
Info: the ICE on GCC 10.3 also occurs with the previous version.

**Comment by sliwowitz** [comment]
I get the same ICE with GCC 12.1.1 (on Fedora 36) both this and the previous version.

**Comment by jschueller** [comment]
also got this one with gcc 12.1.0, archlinux

**Comment by jschueller** [comment]
fixed by 202e2a0990fbef3c0e00dae8335ceb378cd41426, it can be closed I guess

**Comment by SylvainCorlay** [comment]
Note: I have included this patch in the conda recipe for `0.9.1`.

**Comment by SylvainCorlay** [comment]
> I would greatly appreciate a release of libcxxwrap-julia anc CxxWrap.jl including this fix!

cc @barche

**Comment by barche** [comment]
I released a new version (v0.9.2) of libcxxwrap-julia. Not sure if a new release of CxxWrap is also needed, though? The change is only on the C++ side.
